lib/semantics/check-select-stmt.cc

```diff
@@ -0,0 +1,239 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "tools.h"
+#include "../parser/message.h"
+
+namespace Fortran::semantics {
+
+void SelectStmtChecker::Leave(
+    const parser::CaseConstruct &selectCaseConstruct) {
+  const auto &selectCaseStmt{
+      std::get<parser::Statement<parser::SelectCaseStmt>>(
+          selectCaseConstruct.t)};
+  const auto &parsedExpr{
+      std::get<parser::Scalar<parser::Expr>>(selectCaseStmt.statement.t).thing};
+  std::optional<TypeCategory> selectCaseStmtType;
+  bool validCaseStmtType{false};
+  if (const auto *expr{GetExpr(parsedExpr)}) {
```

Comment:
If there is an error in the expression (i.e. GetExpr  returns null), you shouldn't also report an error about the type of the expression. It looks like that's what would happen.

Reply:
Maybe just return in that case?

Reply:
> Maybe just return in that case?

Yes, I would just modify to return if GetExpr returns null instead of issuing an error for the type of the expression.